docs/project-config.md

```diff
@@ -0,0 +1,279 @@
+
+
+## Endpoint
+
+The `endpoint` is being used to tell a platform where it can find the Jovo app.
+
+The default is the Jovo Webhook for local development. The `'${JOVO_WEBHOOK_URL}'` string is automatically replaced with your webhook URL during the `build` process.
+
+```js
+const project = new ProjectConfig({
+  endpoint: '${JOVO_WEBHOOK_URL}',
+  // ...
+});
+```
+
+
+
+## Plugins
+
+Jovo CLI plugins can be used for many use cases. Here are some examples:
+
+* [Platform](./platforms.md) CLI integrations like `AlexaCli` provide the necessary tools to `build` the Alexa interaction model and `deploy` the project to the Alexa developer console.
+* Deployment CLI integrations like `ServerlessCli` allow you to `deploy` your Jovo app to a cloud environment using the Serverless Framework.
+* CLI plugins can even create their own Jovo CLI commands.
+  
+In the next few sections, we'll take a closer look how CLI plugins can be [installed](#installation), [configured](#configuration), and how the Jovo [File Builder](#file-builder) can be used in some cases.
+
+
+### Installation
+
+You can add CLI plugins to the `plugins` array of the project config. Here is an example for Amazon Alexa:
+
+```js
+const { ProjectConfig } = require('@jovotech/cli-core');
+const { AlexaCli } = require('@jovotech/platform-alexa');
+
+// ...
+
+const project = new ProjectConfig({
+  // ...
+  plugins: [
+    new AlexaCli()
+  ],
+});
+```
+
+Many of the platforms or plugins that you add to your [app config](./app-config.md) already come with a CLI plugin. This means that you don't need to install the CLI package separately, if the plugin specific documentation doesn't tell you otherwise.
+
+
+### Configuration
+
+For each CLI plugin, you can add configurations as an array of options that you pass to the plugin when you instantiate it:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  plugins: [
+    new AlexaCli({
+      // Configuration
+    })
+  ],
+});
+```
+
+Each plugin has its own set of options that you can find on the respective documentation pages.
+
+
+### File Builder
+
+For each CLI plugin that hooks into the `build` command, you can use the Jovo File Builder to write specific files into a path of the `build` directory.
+
+You can add the paths and file content to the `files` property of the plugin config:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  plugins: [
+    new AlexaCli({
+      files: {
+        path: {
+          to: {
+            'file.json': 'Hello World!'
+          }
+        }
+      }
+    })
+  ],
+});
+```
+
+The above example creates a file in `build/alexa/path/to/file.json` with the content `Hello World!`.
+
+You can also use this shorthand for nested folders:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  plugins: [
+    new AlexaCli({
+      files: {
+        'path.to.file.json': 'Hello World!'
```

Comment:
This wouldn't work, as a dot is used to mark the file extension. I'll adjust this to use slashes instead.